File: examples/ts/complex/complex.ts
interface GeoLocation {
    lat: number;
    lng: number;
}

interface Address {
    street: string;
    city: string;
    geo: GeoLocation;
}

interface UserProfile {
    id: number;
    username: string;
    isActive: boolean;
    address: Address;
    tags?: string[];
}

interface ProcessSummary {
    totalUsers: number;
    activeUsers: number;
    primaryCity: string;
    details: string[];
}

export function processUsers(users: UserProfile[], requireActive: boolean): ProcessSummary {
    const totalUsers = users.length;
    const activeUsers = users.filter(u => u.isActive);

    const primaryCity = users.length > 0 ? users[0].address.city : "Unknown";

    const details = users.map(u => {
        const tagCount = u.tags ? u.tags.length : 0;
        return `${u.username} (ID: ${u.id}, City: ${u.address.city}, Lat: ${u.address.geo.lat}, Lng: ${u.address.geo.lng}, Tags: ${tagCount})`;
    });

    return {
        totalUsers,
        activeUsers: activeUsers.length,
        primaryCity,
        details
    };
}

File: examples/python/complex/complex.py
from dataclasses import dataclass
from typing import Optional


@dataclass
class UserProfile:
    id: int
    username: str
    is_active: bool
    tags: Optional[list[str]]


def process_users(users: list[UserProfile], require_active: bool) -> str:
    """Process a list of user profiles and return a summary string."""
    total = len(users)
    active = [u for u in users if u.is_active]

    if require_active and len(active) == 0:
        return "Error: No active users found"

    processed = [f"{u.username} (ID: {u.id}, Tags: {len(u.tags or [])})" for u in users]
    return (
        f"Successfully processed {total} users. "
        f"Active: {len(active)}. "
        f"Details: {', '.join(processed)}"
    )
